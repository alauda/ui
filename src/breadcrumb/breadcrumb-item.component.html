<div class="aui-breadcrumb__item">
  <span class="aui-breadcrumb__content"><ng-content></ng-content></span>
  <span class="aui-breadcrumb__separator">
    <ng-container *ngIf="!separatorIcon">{{ separator }}</ng-container>
    <aui-icon
      *ngIf="separatorIcon"
      [icon]="separatorIcon"
    ></aui-icon>
  </span>
</div>
