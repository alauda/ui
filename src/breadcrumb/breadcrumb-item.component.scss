@import '../theme/var';
@import '../theme/mixin';

.aui-breadcrumb__item {
  display: flex;

  .aui-breadcrumb__content {
    max-width: 224px;
    @include text-overflow;
  }
}
