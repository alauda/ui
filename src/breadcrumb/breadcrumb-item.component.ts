import { NgIf } from '@angular/common';
import {
  ChangeDetectionStrategy,
  ChangeDetectorRef,
  Component,
  ViewEncapsulation,
} from '@angular/core';

import { IconComponent } from '../icon/icon.component';

@Component({
    selector: 'aui-breadcrumb-item',
    templateUrl: './breadcrumb-item.component.html',
    styleUrls: ['./breadcrumb-item.component.scss'],
    changeDetection: ChangeDetectionStrategy.OnPush,
    encapsulation: ViewEncapsulation.None,
    preserveWhitespaces: false,
    imports: [NgIf, IconComponent]
})
export class BreadcrumbItemComponent {
  get separator() {
    return this._separator;
  }

  set separator(val) {
    this._separator = val;
    this.cdr.markForCheck();
  }

  get separatorIcon() {
    return this._separatorIcon;
  }

  set separatorIcon(val) {
    this._separatorIcon = val;
    this.cdr.markForCheck();
  }

  private _separator = '/';
  private _separatorIcon = '';

  constructor(private readonly cdr: ChangeDetectorRef) {}
}
