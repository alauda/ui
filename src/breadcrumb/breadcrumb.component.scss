@import '../theme/var';
@import '../theme/mixin';

.aui-breadcrumb {
  display: flex;
  align-items: center;
  @include text-set(m, secondary);

  &__separator {
    margin: 0 use-var(spacing-s);
  }

  aui-breadcrumb-item:last-child #{&}__separator {
    display: none;
  }
}
