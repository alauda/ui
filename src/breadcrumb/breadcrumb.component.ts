import {
  AfterContentInit,
  ChangeDetectionStrategy,
  Component,
  ContentChildren,
  Input,
  QueryList,
  ViewEncapsulation,
} from '@angular/core';

import { BreadcrumbItemComponent } from './breadcrumb-item.component';

@Component({
  selector: 'aui-breadcrumb',
  templateUrl: './breadcrumb.component.html',
  styleUrls: ['./breadcrumb.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
  encapsulation: ViewEncapsulation.None,
  preserveWhitespaces: false,
  standalone: true,
})
export class BreadcrumbComponent implements AfterContentInit {
  @Input()
  get separator() {
    return this._separator;
  }

  set separator(val) {
    if (val === this._separator) {
      return;
    }
    this._separator = val;
    if (this.items) {
      this.items.forEach(item => {
        item.separator = val;
      });
    }
  }

  @Input()
  get separatorIcon() {
    return this._separatorIcon;
  }

  set separatorIcon(val) {
    if (val === this._separatorIcon) {
      return;
    }
    this._separatorIcon = val;
    if (this.items) {
      this.items.forEach(item => {
        item.separatorIcon = val;
      });
    }
  }

  @ContentChildren(BreadcrumbItemComponent)
  items: QueryList<BreadcrumbItemComponent>;

  private _separator = '/';
  private _separatorIcon = '';

  ngAfterContentInit() {
    this.items.forEach(item => {
      item.separator = this.separator;
      item.separatorIcon = this.separatorIcon;
    });
  }
}
