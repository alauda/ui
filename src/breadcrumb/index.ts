export * from './breadcrumb.component';
export * from './breadcrumb.module';
export * from './breadcrumb-item.component';
