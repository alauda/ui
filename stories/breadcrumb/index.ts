import { storiesOf } from '@storybook/angular';

import { BreadcrumbModule } from '@alauda/ui';

storiesOf('Breadcrumb', module).add('breadcrumb', () => ({
  moduleMetadata: {
    imports: [BreadcrumbModule],
  },
  template: /* HTML */ `
    <aui-breadcrumb>
      <aui-breadcrumb-item><a href="/">storybook</a></aui-breadcrumb-item>
      <aui-breadcrumb-item><a href="./">breadcrumb</a></aui-breadcrumb-item>
      <aui-breadcrumb-item>default</aui-breadcrumb-item>
    </aui-breadcrumb>
    <aui-breadcrumb separator="|">
      <aui-breadcrumb-item><a href="/">storybook</a></aui-breadcrumb-item>
      <aui-breadcrumb-item><a href="./">breadcrumb</a></aui-breadcrumb-item>
      <aui-breadcrumb-item>default</aui-breadcrumb-item>
    </aui-breadcrumb>
    <aui-breadcrumb separatorIcon="angle_right">
      <aui-breadcrumb-item><a href="/">storybook</a></aui-breadcrumb-item>
      <aui-breadcrumb-item><a href="./">breadcrumb</a></aui-breadcrumb-item>
      <aui-breadcrumb-item>default</aui-breadcrumb-item>
    </aui-breadcrumb>
  `,
}));
