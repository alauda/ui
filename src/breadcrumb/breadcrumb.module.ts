import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { IconModule } from '../icon';

import { BreadcrumbItemComponent } from './breadcrumb-item.component';
import { BreadcrumbComponent } from './breadcrumb.component';

@NgModule({
  imports: [CommonModule, IconModule],
  declarations: [BreadcrumbComponent, BreadcrumbItemComponent],
  exports: [BreadcrumbComponent, BreadcrumbItemComponent],
})
export class BreadcrumbModule {}
