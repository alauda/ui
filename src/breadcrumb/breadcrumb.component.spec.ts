import { Component } from '@angular/core';
import { TestBed } from '@angular/core/testing';

import { BreadcrumbModule } from './breadcrumb.module';

describe('BreadcrumbComponent', () => {
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [BreadcrumbModule, TestComponent],
    });
  });

  it('should render correct template', () => {
    const fixture = TestBed.createComponent(TestComponent);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});

@Component({
  template: `
    <aui-breadcrumb>
      <aui-breadcrumb-item><a href="/">storybook</a></aui-breadcrumb-item>
      <aui-breadcrumb-item><a href="./">breadcrumb</a></aui-breadcrumb-item>
      <aui-breadcrumb-item>default</aui-breadcrumb-item>
    </aui-breadcrumb>
    <aui-breadcrumb separator="|">
      <aui-breadcrumb-item><a href="/">storybook</a></aui-breadcrumb-item>
      <aui-breadcrumb-item><a href="./">breadcrumb</a></aui-breadcrumb-item>
      <aui-breadcrumb-item>default</aui-breadcrumb-item>
    </aui-breadcrumb>
    <aui-breadcrumb separatorIcon="angle_right">
      <aui-breadcrumb-item><a href="/">storybook</a></aui-breadcrumb-item>
      <aui-breadcrumb-item><a href="./">breadcrumb</a></aui-breadcrumb-item>
      <aui-breadcrumb-item>default</aui-breadcrumb-item>
    </aui-breadcrumb>
  `,
  standalone: true,
  imports: [BreadcrumbModule],
})
class TestComponent {}
