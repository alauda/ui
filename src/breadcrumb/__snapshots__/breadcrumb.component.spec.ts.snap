// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbComponent should render correct template 1`] = `
<ng-component>
  <aui-breadcrumb>
    <div
      class="aui-breadcrumb"
    >
      <aui-breadcrumb-item>
        <div
          class="aui-breadcrumb__item"
        >
          <span
            class="aui-breadcrumb__content"
          >
            <a
              href="/"
            >
              storybook
            </a>
          </span>
          <span
            class="aui-breadcrumb__separator"
          >
            
            
            /
            
          </span>
        </div>
      </aui-breadcrumb-item>
      <aui-breadcrumb-item>
        <div
          class="aui-breadcrumb__item"
        >
          <span
            class="aui-breadcrumb__content"
          >
            <a
              href="./"
            >
              breadcrumb
            </a>
          </span>
          <span
            class="aui-breadcrumb__separator"
          >
            
            
            /
            
          </span>
        </div>
      </aui-breadcrumb-item>
      <aui-breadcrumb-item>
        <div
          class="aui-breadcrumb__item"
        >
          <span
            class="aui-breadcrumb__content"
          >
            default
          </span>
          <span
            class="aui-breadcrumb__separator"
          >
            
            
            /
            
          </span>
        </div>
      </aui-breadcrumb-item>
    </div>
  </aui-breadcrumb><aui-breadcrumb
    separator="|"
  >
    <div
      class="aui-breadcrumb"
    >
      <aui-breadcrumb-item>
        <div
          class="aui-breadcrumb__item"
        >
          <span
            class="aui-breadcrumb__content"
          >
            <a
              href="/"
            >
              storybook
            </a>
          </span>
          <span
            class="aui-breadcrumb__separator"
          >
            
            
            |
            
          </span>
        </div>
      </aui-breadcrumb-item>
      <aui-breadcrumb-item>
        <div
          class="aui-breadcrumb__item"
        >
          <span
            class="aui-breadcrumb__content"
          >
            <a
              href="./"
            >
              breadcrumb
            </a>
          </span>
          <span
            class="aui-breadcrumb__separator"
          >
            
            
            |
            
          </span>
        </div>
      </aui-breadcrumb-item>
      <aui-breadcrumb-item>
        <div
          class="aui-breadcrumb__item"
        >
          <span
            class="aui-breadcrumb__content"
          >
            default
          </span>
          <span
            class="aui-breadcrumb__separator"
          >
            
            
            |
            
          </span>
        </div>
      </aui-breadcrumb-item>
    </div>
  </aui-breadcrumb><aui-breadcrumb
    separatoricon="angle_right"
  >
    <div
      class="aui-breadcrumb"
    >
      <aui-breadcrumb-item>
        <div
          class="aui-breadcrumb__item"
        >
          <span
            class="aui-breadcrumb__content"
          >
            <a
              href="/"
            >
              storybook
            </a>
          </span>
          <span
            class="aui-breadcrumb__separator"
          >
            
            
            <aui-icon>
              
              
              <svg
                class="aui-icon aui-icon-angle_right"
              >
                <use
                  xlink:href="#aui-icon-angle_right"
                />
              </svg>
              
              
            </aui-icon>
          </span>
        </div>
      </aui-breadcrumb-item>
      <aui-breadcrumb-item>
        <div
          class="aui-breadcrumb__item"
        >
          <span
            class="aui-breadcrumb__content"
          >
            <a
              href="./"
            >
              breadcrumb
            </a>
          </span>
          <span
            class="aui-breadcrumb__separator"
          >
            
            
            <aui-icon>
              
              
              <svg
                class="aui-icon aui-icon-angle_right"
              >
                <use
                  xlink:href="#aui-icon-angle_right"
                />
              </svg>
              
              
            </aui-icon>
          </span>
        </div>
      </aui-breadcrumb-item>
      <aui-breadcrumb-item>
        <div
          class="aui-breadcrumb__item"
        >
          <span
            class="aui-breadcrumb__content"
          >
            default
          </span>
          <span
            class="aui-breadcrumb__separator"
          >
            
            
            <aui-icon>
              
              
              <svg
                class="aui-icon aui-icon-angle_right"
              >
                <use
                  xlink:href="#aui-icon-angle_right"
                />
              </svg>
              
              
            </aui-icon>
          </span>
        </div>
      </aui-breadcrumb-item>
    </div>
  </aui-breadcrumb>
</ng-component>
`;
